itest/universe_pagination_test.go

```diff
@@ -0,0 +1,246 @@
+package itest
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/taprpc"
+	"github.com/lightninglabs/taproot-assets/taprpc/mintrpc"
+	unirpc "github.com/lightninglabs/taproot-assets/taprpc/universerpc"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// testPageSize is the page size to use when fetching data from the
+	// universe rpc. We use a small page size to test pagination.
+	testPageSize = 42
+
+	// testPageSizeSmall is the page size to use when fetching data from the
+	// universe rpc. We use a small page size to test pagination.
+	testPageSizeSmall = 2
+
+	// testGroupSize is the size of the asset group we mint in the
+	// testUniversePaginationSimple test.
+	testGroupSize = 79
+)
+
+func testUniversePaginationSimple(t *harnessTest) {
+	mintSize := 255
+	timeout := defaultWaitTimeout
+
+	// If we create a second tapd instance and sync the universe state,
+	// the synced tree should match the source tree.
+	bob := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, nil,
+	)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	mintBatches := func(reqs []*mintrpc.MintAssetRequest) []*taprpc.Asset {
+		return MintAssetsConfirmBatch(
+			t.t, t.lndHarness.Miner.Client, t.tapd, reqs,
+			WithMintingTimeout(timeout),
+		)
+	}
+
+	imageMetadataBytes := GetImageMetadataBytes(t.t, ImageMetadataFileName)
+
+	mintBatchAssetsTest(
+		t.t, t.tapd, bob, t.tapd.rpcHost(), mintSize, mintBatches,
+		imageMetadataBytes, timeout,
+	)
+}
+
+// mintBatchAssetsTest mints many assets
+func mintBatchAssetsTest(
```

Comment:
IIUC this was lifted from another test, so no need to fully copy/paste, may need some intermediate refactor.

Reply:
the func body is changed so that multiple groups are created, this func creates multiple batches